src/partitioner/stream/Partition.cpp

```diff
@@ -0,0 +1,152 @@
+        return this->edgeList[vertex];
+    } else {
+        return {};
+    }
+}
+
+// The number of edges, the cardinality of E, is called the size of graph and denoted by |E|. We usually use m to denote
+// the size of G.
+double Partition::getEdgesCount() {
+    double total = 0;
+    std::set<std::string> uniqueEdges;
+    for (auto edge : this->edgeList) {
+        for (auto vertext : edge.second) {
+            uniqueEdges.insert(edge.first + vertext);
+        }
+    }
+    return uniqueEdges.size();
+}
+
+// The number of vertices, the cardinality of V, is called the order of graph and devoted by |V|. We usually use n to
+// denote the order of G.
+double Partition::getVertextCount() {
+    double edgeListVetices = this->edgeList.size();
+    double edgeCutVertices = 0;
+    for (size_t i = 0; i < this->numberOfPartitions; i++) {
+        for (auto edge : this->edgeCuts[i]) {
+            bool isExistInEdgeList = this->edgeList.find(edge.first) != this->edgeList.end();
+            if (!isExistInEdgeList) {
+                edgeCutVertices += 1;
+            }
+        }
+    }
+    return edgeListVetices + edgeCutVertices;
+}
+
+template <typename Out>
+void Partition::_split(const std::string &s, char delim, Out result) {
+    std::stringstream ss(s);
+    std::string item;
+    while (std::getline(ss, item, delim)) {
+        *(result++) = item;
+    }
+}
+
+std::vector<std::string> Partition::_split(const std::string &s, char delim) {
+    std::vector<std::string> elems;
+    Partition::_split(s, delim, std::back_inserter(elems));
+    return elems;
+}
+
+void Partition::addToEdgeCuts(std::string resident, std::string foreign, int partitionId) {
+    if (partitionId < this->numberOfPartitions) {
+        auto exsistResidentVertex = this->edgeCuts[partitionId].find(resident);
+        if (exsistResidentVertex != this->edgeCuts[partitionId].end()) {
+            this->edgeCuts[partitionId][resident].insert(foreign);
+        } else {
+            this->edgeCuts[partitionId][resident] = std::unordered_set<std::string>({foreign});
+        }
+    }
+}
+
+long Partition::edgeCutsCount() {
+    long total = 0;
+    for (auto partition : this->edgeCuts) {
+        for (auto edgeCuts : partition) {
+            total += edgeCuts.second.size();
+        }
+    }
+    return total;
+}
+
+float Partition::edgeCutsRatio() { return this->edgeCutsCount() / (this->getEdgesCount() + this->edgeCutsCount()); }
+
+void Partition::printEdgeCuts() {
+    std::cout << "Printing edge cuts of " << id << " partition" << std::endl;
+    for (auto partition : this->edgeCuts) {
+        for (auto edgeList : partition) {
+            std::cout << edgeList.first << " ____" << std::endl;
+            for (std::string vertext : edgeList.second) {
+                std::cout << "\t| ---> " << vertext << std::endl;
+            }
+            std::cout << "\n" << std::endl;
+        }
+    }
+}
+
+void Partition::printEdges() {
```

Comment:
The same comment on `cout` applies here too.

Reply:
Fixed in https://github.com/miyurud/jasminegraph/pull/122/commits/bdd734e88390169ddc9e44c74030d1c1b170d0ae

Reply:
Done